### PR TITLE
[SDK-1405] Added QR Code Methods

### PR DESCRIPTION
**Pull request by nsingh-branch**
Created BranchQRCode with methods to generate a Branch QR code as data or an image. Also created BranchQRCode cache to cache recently created QR codes. This PR includes tests and a new TestBed button to generate a QR code as well.

### CODE REVIEW

**Review by gdeluna-branch — APPROVED** [review]

File: Branch/BranchQRCode.m [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+NSString *buoTitle;
+CIImage *qrCodeImage;
+
+- (void) setMargin:(NSNumber *)margin {
+    if (margin.intValue > 20) {
+        margin = @(20);
+        BNCLogWarning(@"Margin was reduced to the maximum of 20.");
+    }
+    if (margin.intValue < 1) {
```
Comment (NidhiDixit09) [inline Branch/BranchQRCode.m:24]: Check should be  if (margin.intValue < 0) {

File: Branch/BranchQRCode.m [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    _margin = margin;
+}
+
+- (void) setWidth:(NSNumber *)width {
+    if (width.intValue > 2000) {
+        width = @(2000);
+        BNCLogWarning(@"Width was reduced to the maximum of 2000.");
+    }
+    if (width.intValue < 300) {
```
Comment (NidhiDixit09) [inline Branch/BranchQRCode.m:36]: Check should be for 500 instead of 300

File: Branch/BranchQRCode.m [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+#import "BranchQRCode.h"
+#import "Branch.h"
+#import "BNCQRCodeCache.h"
+
+@implementation BranchQRCode
+
+NSString *buoTitle;
+CIImage *qrCodeImage;
+
+- (void) setMargin:(NSNumber *)margin {
```
Comment (NidhiDixit09) [inline Branch/BranchQRCode.m:19]: There is no default value of margin and width ( no init function). If user doesn't set margin and width, what will be behavior ? QR code will generate code automatically with some server side default values ?

**Review by NidhiDixit09 — COMMENTED** [review]
@nsingh-branch I have added some comments/queries. Overall code looks good to me

File: Branch/BranchQRCode.m (hunk 3 above)
Comment (nsingh-branch) [reply Branch/BranchQRCode.m:19]: Yeah, there are server-side defaults so all of the QR code settings are optional.

File: Branch/BranchQRCode.m (hunk 2 above)
Comment (nsingh-branch) [reply Branch/BranchQRCode.m:36]: Updated the comment to reflect the minimum.

File: Branch/BranchQRCode.m (hunk 1 above)
Comment (nsingh-branch) [reply Branch/BranchQRCode.m:24]: Updated the comment to reflect the minimum.

**Review by NidhiDixit09 — APPROVED** [review]

**Pull request merged**